### DISCUSSION TITLE
Ability to add components

### DISCUSSION THREAD

**Discussion by Wulfheart** [spatie/laravel-package-tools#12] [Ideas]
Currently it is not possible to register package blade components. Am I correct?

**Comment by freekmurze** [comment]
You can do it in the same way you do it in regular Laravel apps

**Reply by Wulfheart** [reply]
Wait.. So no registering it at the [boot method](https://laravel.com/docs/8.x/blade#manually-registering-components)?
Simply adding a `components` to the `views` folder?

**Comment by thibodelanghe** [comment]
Hi @freekmurze, thanks for this package. It'll clean up our service providers quite a bit, but i have the following question. Isn't is more logical that the Components directory is in the src folder instead of 1 level above it or do i miss something?
https://github.com/spatie/laravel-package-tools/blob/master/src/PackageServiceProvider.php#L113

**Reply by freekmurze** [reply]
Yeah, that's strange. I don't think I review the PR correctly that added that. I think components should go in `src/Http/Component/` 

I accept a PR that will load the components from directory additionally.

**Reply by thibodelanghe** [reply]
Hi freek, i've submitted a PR. https://github.com/spatie/laravel-package-tools/pull/35

**Comment by dominik-eller** [comment]
In am not getting my component to work as expected. The components view file gets loaded but the corresponding component class is not. So I am having a component and am not able to pass variables to it.

I am having

      src/Components/Component.php
      resources/views/components/component.blade.php

and I am using

      ->hasViewComponents('vendor', Component::class)
also tried

      ->hasViewComponent('vendor', Component::class)

I am loading the component with

      <x-vendor::component />

I have no idea why by Component.php class is not loaded to pass the data into my view. I compared my implementation with https://github.com/spatie/laravel-health and there is no difference.

I am using a fresh Laravel 11 installation...

**Reply by dominik-eller** [reply]
The following change did it:

       <x-vendor-component />
